common/lib/xmodule/xmodule/course_module.py

```diff
@@ -269,6 +269,44 @@ def get_available_providers():
     return available_providers
 
 
+class DiscussionTopicMapping(Dict):
+    """
+    DiscussionTopicMapping field, which includes validation of the topic id
+    """
+
+    _default = {}
+
+    def from_json(self, value):
+        """
+        Return python dict type. Performs validation on id key for each discussion topic
+        """
+        if value is None:
+            return None
+        """
+        if statement checks to make sure value is a dict,
+        each topic in dict has a key id and the id value only has valid characters
+        """
+        if isinstance(value, dict):
+            for key in value:
+                if "id" in value[key].keys():
+                    if not self._validate_topic_id_value(value[key]["id"]):
+                        raise ValueError("The only special characters that are supported in the id key are underscore, hyphen, and period.")
+                else:
+                    raise ValueError("All topics need a id key")
+        else:
+            raise TypeError('Value stored in a Dict must be None or a dict, found %s' % type(value))
```

Comment:
Can we make this more descriptive as well? If you see this in the logs out of context it doesn't really help debug anything. Same question on if any users would see this or is it only for log files?

Reply:
I think this particular error will not be seen by user, will check to make sure.